File: lecture11/exercise11-B/template.cpp
// Exercise 11-B: Passcode Cracking

#include <iostream>
#include <queue>
#include <string>
#include <unordered_map>
#include <unordered_set>

using namespace std;

// Structures

struct Graph {
    unordered_map<char, unordered_set<char>> edges;
    unordered_map<char, int>		     degrees;
};

// Functions

Graph read_graph() {
    // TODO
}

string topological_sort(Graph &graph) {
    // TODO
}

// Main Execution

int main(int argc, char *argv[]) {
    auto graph = read_graph();
    auto code  = topological_sort(graph);

    cout << code << endl;
    return EXIT_SUCCESS;
}

File: lecture11/exercise11-B/template.py
#!/usr/bin/env python3

# Exercise 11-B: Passcode Cracking

import collections
import sys

# Graph structure

Graph = collections.namedtuple('Graph', 'edges degrees')

# Read Graph

def read_graph() -> Graph:
    # TODO
    pass

# Topological Sort

def topological_sort(graph: Graph) -> list[int]:
    # TODO
    pass

# Main Execution

def main() -> None:
    graph = read_graph()
    code  = topological_sort(graph)

    print(''.join(map(str, code)))

if __name__ == '__main__':
    main()
